### PR TITLE
fix: Aligns top level tabs when native filters are enabled

### LINKED ISSUE #14973: [native-filters]expanding filter bar should push top-level tabs along with main dashboard to the right
https://user-images.githubusercontent.com/67837651/120715264-3ab59e00-c479-11eb-9ed0-aa98238b1b85.mov

Screen Shot 2021-06-03 at 2 35 29 PM

@michael-s-molina @villebro 🙏

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Fixes https://github.com/apache/superset/issues/14973

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
See the original issue for before screenshots.

https://user-images.githubusercontent.com/70410625/120821221-a6584380-c52b-11eb-82fa-40c3868c7631.mp4

@villebro @junlincc

### TESTING INSTRUCTIONS
1 - Enable the native filters
2 - Open a dashboard with top-level tabs
3 - Expand the native filters
4 - The top-level tabs should be aligned with the dashboard content

### ADDITIONAL INFORMATION
- [x] Has associated issue:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by kgabryje** [comment]
What do you think about making it look like this? 
image

I'm not sure if I like that empty space on the left side of the bar with top level tabs

File: superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx [hunk 1]
```diff
@@ -161,6 +162,11 @@ const DashboardBuilder: FC<DashboardBuilderProps> = () => {
     (hideDashboardHeader ? 0 : HEADER_HEIGHT) +
     (topLevelTabs ? TABS_HEIGHT : 0);
 
+  const tabsPaddingLeft =
+    nativeFiltersEnabled && !editMode && dashboardFiltersOpen
+      ? NATIVE_FILTER_BAR_WIDTH + 20
+      : 8;
```
Comment (kgabryje) [inline superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx:168]: Can we use theme variables here? `{theme.gridUnit * 2}px`

File: superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx [hunk 2]
```diff
@@ -55,6 +55,7 @@ import DashboardContainer from './DashboardContainer';
 
 const TABS_HEIGHT = 47;
 const HEADER_HEIGHT = 67;
+const NATIVE_FILTER_BAR_WIDTH = 255;
```
Comment (kgabryje) [inline superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx:58]: Why 255 instead of 250?

**Comment by michael-s-molina** [comment]
> What do you think about making it look like this?
> image
> 
> I'm not sure if I like that empty space on the left side of the bar with top level tabs

I also think this is the best approach but it seems that we have a requirement from a client asking to nest the filters inside a top-level tab.  @suddjian @junlincc can you check if we can change this requirement?

File: superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx (hunk 2 above)
Comment (michael-s-molina) [reply superset-frontend/src/dashboard/components/DashboardBuilder/DashboardBuilder.tsx:58]: To fix the inner margin-right of the filter values. They have a fixed width.

File: superset-frontend/src/dashboard/components/StickyVerticalBar.tsx [hunk 3]
```diff
@@ -79,6 +80,7 @@ export const StickyVerticalBar: React.FC<SVBProps> = ({
           distanceFromTop: number;
         }) => (
           <Contents
+            id="sticky-vertical-bar"
```
Comment (kgabryje) [inline superset-frontend/src/dashboard/components/StickyVerticalBar.tsx:83]: Do we need that id here?

**Comment by kgabryje** [comment]
> I also think this is the best approach but it seems that we have a requirement from a client asking to nest the filters inside a top-level tab. @suddjian @junlincc can you check if we can change this requirement?

I feel like that's a case that we discussed some time ago - that we can't treat native filters like filter box with slightly different UI. I think that native filters are "global" to the dashboard by design and the panel doesn't belong to a single top level tab. Scoping of some filters to a tab have been addressed by highlighting tabs and hiding filters that are out of scope, so we don't need to nest the native filters panel to indicate that it belongs to some tab. Though let's wait for an opinion from product point of view  🙂 cc @junlincc

**Comment by villebro** [comment]
> > I also think this is the best approach but it seems that we have a requirement from a client asking to nest the filters inside a top-level tab. @suddjian @junlincc can you check if we can change this requirement?
> 
> I feel like that's a case that we discussed some time ago - that we can't treat native filters like filter box with slightly different UI. I think that native filters are "global" to the dashboard by design and the panel doesn't belong to a single top level tab. Scoping of some filters to a tab have been addressed by highlighting tabs and hiding filters that are out of scope, so we don't need to nest the native filters panel to indicate that it belongs to some tab. Though let's wait for an opinion from product point of view 🙂 cc @junlincc

The possibility to place native filters on the dashboard layout (similar to Filter Box) is outside the scope of Native Filters - those that want to do that can enable the `DASHBOARD_FILTERS_EXPERIMENTAL` feature flag. While placing a filter component close to charts within scope can feel intuitive, I personally feel the new in/out of scope logic introduced by #14933 provides a better UX and makes it easier to quickly check which filters are available and in scope.

**Comment by suddjian** [comment]
@michael-s-molina I agree. The design with the filters outside of tabs better communicates to the user how the filters interact with the rest of the application, so we should do that.

**Comment by junlincc** [comment]
image


@michael-s-molina please proceed this change, this is much better. 🙏 thank you all for offering suggestions.

**Comment by junlincc** [comment]
@michael-s-molina Hi Michael, dashboard top right tab and dashboard are not aligned perfectly especially on resize. could we address in this or the new design pr? 

https://user-images.githubusercontent.com/67837651/121069001-d3059880-c781-11eb-8ad2-66768897f7c8.mov

**Comment by michael-s-molina** [comment]
I'm closing this PR and opening a new one with the proposed solution.

**Pull request closed without merging**